src/k8s/pkg/k8sd/setup/kubelet.go

```diff
@@ -21,8 +21,26 @@ var kubeletTLSCipherSuites = []string{
 	"TLS_RSA_WITH_AES_256_GCM_SHA384",
 }
 
-// Kubelet configures kubelet on the local node.
-func Kubelet(snap snap.Snap, hostname string, nodeIP net.IP, clusterDNS string, clusterDomain string, cloudProvider string) error {
+var kubeletControlPlaneLabels = []string{
+	"node-role.kubernetes.io/control-plane=",
+}
+
+var kubeletWorkerLabels = []string{
+	"node-role.kubernetes.io/worker=",
+}
+
+// KubeletControlPlane configures kubelet on a control plane node.
+func KubeletControlPlane(snap snap.Snap, hostname string, nodeIP net.IP, clusterDNS string, clusterDomain string, cloudProvider string) error {
+	return kubelet(snap, hostname, nodeIP, clusterDNS, clusterDomain, cloudProvider, nil, append(kubeletControlPlaneLabels, kubeletWorkerLabels...))
```

Comment:
Do we really want to append the worker labels to the control plane?

Reply:
So, currently our control plane are also workers. Optionally, in the future, we could have nodes that are "control-plane" only. For these, we would have to take care of not including the worker labels indeed.

For now, if a workload requires the presence of a worker node to run, we still want a single-node k8s cluster to support it, so we just include the worker labels as well

Reply:
Makes sense!